apollo-router/src/plugins/cache/invalidation.rs

```diff
@@ -36,33 +71,53 @@ impl Invalidation {
         let mut notify = Notify::new(None, None, None);
         let (handle, _b) = notify.create_or_subscribe(InvalidationTopic, false).await?;
         let enabled = storage.is_some();
-        if let Some(storage) = storage {
+        if let Some(storage) = storage.clone() {
             let h = handle.clone();
 
             tokio::task::spawn(async move { start(storage, h.into_stream()).await });
         }
-        Ok(Self { enabled, handle })
+        Ok(Self {
+            enabled,
+            handle,
+            // storage,
+        })
     }
 
     pub(crate) async fn invalidate(
         &mut self,
         origin: InvalidationOrigin,
         requests: Vec<InvalidationRequest>,
-    ) -> Result<(), BoxError> {
+    ) -> Result<u64, BoxError> {
         if self.enabled {
             let mut sink = self.handle.clone().into_sink();
-            sink.send((origin, requests)).await.map_err(|e| e.message)?;
-        }
+            let (response_tx, mut response_rx) = broadcast::channel(1);
+            sink.send((requests, origin, response_tx))
+                .await
+                .map_err(|e| e.message)?;
```

Comment:
Could this be the cause of the BAD_REQUEST that we are seeing in performance testing?

Reply:
I think so